packages/fxa-auth-server/test/local/metrics/amplitude.js

```diff
@@ -400,6 +400,44 @@ describe('metrics/amplitude', () => {
       });
     });
 
+    describe('account.verified, newsletters is undefined', () => {
+      beforeEach(() => {
+        return amplitude('account.verified', mocks.mockRequest({}), {
+          newsletters: undefined
+        });
+      });
+
+      it('did not call log.error', () => {
+        assert.equal(log.error.callCount, 0);
+      });
+
+      it('called log.amplitudeEvent correctly', () => {
+        assert.equal(log.amplitudeEvent.callCount, 1);
+        const args = log.amplitudeEvent.args[0];
+        assert.equal(args[0].event_type, 'fxa_reg - email_confirmed');
+        assert.equal(args[0].user_properties.newsletters, undefined);
```

Comment:
assert.equal against undefined could match any falsey value, can't it? Should you use assert.isUndefined here? Or, should we pass an empty array?

Reply:
Fwiw I think we need to pass an empty array for this case, because of the way `user_properties` are carried forward over time. Otherwise the absence of any value will just revert to whatever the previous value was.

Reply:
(sorry @vbudhram, I realise now I should have mentioned that in the `fxa-shared` PR)

Reply:
Actually, looking at the current code in `fxa-shared`, it will be fine as long as it gets passed in as an empty array in the fully-unsubscribed case. Worth having an extra test here that passes it in as an empty array?

Reply:
(and none of this is relevant to Shane's original comment, sorry; I agree that `isUndefined` is a better assertion to use fwiw)

Reply:
> (and none of this is relevant to Shane's original comment, sorry; I agree that isUndefined is a better assertion to use fwiw)

It's all relevant! Passing an empty array makes sense to me too.

Reply:
Sold! Updated this to pass an empty array.